### PR TITLE
Discover suite tests using parent configuration parameters

### PR DESCRIPTION
**Pull request by mpkorstanje**
## Overview

From a practical perspective the expectation is that a `@Suite` annotated class uses the same configuration parameters when discovering tests as the discovery request used to discover the suite.

For example:

When executing a suite with the the `ConsoleLauncher` providing a configuration via `--config=key=value` should have the same effect as configuring this value implicitly as an environment variable via `-Dvalue=key`. However when suites are involved this is not the case. Rather implicit variables are considered while explicit values are ignored.

This surprising to say the least.

This behavior occurs because the `junit-platform-suite-engine` creates a new discovery request - and this includes the implicit configuration by default.

There is also a use case where the suite should neither inherit configuration nor use any implicit configuration values. (e.g. the execution of tcks  #2594).

This PR changes the `junit-platform-suite-engine` so as to only discover tests using the explicitly provided parameters and the parameters from the discovery request used to discover the the suite engine. 

This behaviour can be disabled by `@DisableParentConfigurationParameters`. If disabled, only the explicitly provided parameters will be used. 

This also means that a suite will never use implicit parameters (unless of course the parent discovery request used them).

Note this replaces: https://github.com/junit-team/junit5/pull/2618

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

**Comment by sbrannen** [comment]
Tentatively slated for 5.8 M2 for _team discussion_

**Comment by sbrannen** [comment]
This PR is currently blocked until #2680 has been merged into `main`.

Since the two PRs modify the same classes, this PR will need to be rebased on `main` once #2680 has been merged.

----

@mpkorstanje, I will likely merge #2680 in the coming days since we plan to release 5.8 M2 within a week or so. Do you think you'll have time to rebase this on `main` within that time frame?

**Comment by mpkorstanje** [comment]
Unless you are expecting major changes I can rebase on #2680 now. That should make rebasing on main after merge easier.

**Comment by mpkorstanje** [comment]
Rebasing on `main` seems to have resolved the conflicts with #2680.

**Comment by sbrannen** [comment]
> Rebasing on `main` seems to have resolved the conflicts with #2680.

Great. Thanks for taking care of the rebase!

**Review by marcphilipp — APPROVED** [review]
Nice work! 👍

**Review by sormuras — APPROVED** [review]

**Comment by marcphilipp** [comment]
Thanks, @mpkorstanje! 👍

**Pull request merged**
